tarwriter.go

```diff
@@ -50,8 +58,38 @@ func (w *TarWriter) writeFile(f File, fpath string) error {
 	return nil
 }
 
+func validateTarFilePath(baseDir, fpath string) bool {
+	// Ensure the filepath has no ".", "..", etc within the known root directory.
+	fpath = path.Clean(fpath)
+
+	// If we have a non-empty baseDir, check if the filepath starts with baseDir.
+	// If not, we can exclude it immediately. For 'ipfs get' and for the gateway,
+	// the baseDir would be '{cid}.tar'.
+	if baseDir != "" && !strings.HasPrefix(path.Clean(fpath), baseDir) {
+		return false
+	}
+
+	// If there is no defined baseDir, i.e., if there's multiple files in the
+	// root, check if paths contain elements that would otherwise make them fall
+	// outside the root.
+	if strings.Contains(fpath, "..") {
```

Comment:
I think, It should be startwith and not contains because you clean it up already.